src/main/java/org/gridsuite/modification/server/modifications/NetworkStoreListener.java

```diff
@@ -267,38 +267,58 @@ private void flushEquipmentInfos() {
-            resImpacts.add(CollectionElementImpact.builder()
+    private List<AbstractBaseImpact> flushNetworkImpacts() {
+        if (isAllNetworkImpacted()) {
+            return List.of(CollectionElementImpact.builder()
                     .elementType(IdentifiableType.SUBSTATION)
                     .build());
         }
 
-        // then filter those DELETION impacts for the next part and the collection impact computation
-        Set<SimpleElementImpact> filteredImpacts = impacts.stream()
-            .filter(i -> i.getSimpleImpactType() != SimpleImpactType.DELETION)
-            .collect(Collectors.toSet());
-        Set<IdentifiableType> impactTypes = filteredImpacts.stream()
-            .map(i -> i.getElementType()).collect(Collectors.toSet());
-
-        impactTypes.forEach(type -> {
-            // get SimpleElementimpacts of this equipment type
-            Set<SimpleElementImpact> typedNetworkImpacts = filteredImpacts.stream().filter(i -> type == i.getElementType()).collect(Collectors.toSet());
-            // compare using a threshold (ex: if nbLoadImpacts >= 50)
-            if (typedNetworkImpacts.size() >= collectionThreshold) {
-                resImpacts.add(CollectionElementImpact.builder()
-                    .elementType(type)
+        return new ArrayList<>(reduceNetworkImpacts());
+    }
+
+    private boolean isAllNetworkImpacted() {
+        Set<String> allImpactedSubstationsIds = simpleImpacts.stream().flatMap(i -> i.getSubstationIds().stream()).collect(Collectors.toSet());
+        return allImpactedSubstationsIds.size() >= collectionThreshold;
+    }
+
+    private Set<AbstractBaseImpact> reduceNetworkImpacts() {
+        Set<AbstractBaseImpact> reducedImpacts = new HashSet<>();
+        Set<String> impactedSubstationsIds = new HashSet<>();
+
+        // Impacts type collection
+        Arrays.stream(IdentifiableType.values()).forEach(elementType -> {
+            Set<SimpleElementImpact> impacts = getSimpleImpacts(elementType);
+            if (impacts.size() >= collectionThreshold) {
+                reducedImpacts.add(CollectionElementImpact.builder()
+                    .elementType(elementType)
                     .build());
             } else {
-                // otherwise add the Simple Element impacts
-                resImpacts.addAll(typedNetworkImpacts);
+                impactedSubstationsIds.addAll(impacts.stream().flatMap(i -> i.getSubstationIds().stream()).collect(Collectors.toSet()));
             }
         });
-        return resImpacts;
+
+        // Impacts type simple for substation only
+        reducedImpacts.addAll(impactedSubstationsIds.stream().map(id ->
+            SimpleElementImpact.builder()
+                .simpleImpactType(SimpleImpactType.MODIFICATION)
+                .elementType(IdentifiableType.SUBSTATION)
+                .elementId(id)
+                .substationIds(Set.of(id))
+                .build()
+        ).collect(Collectors.toSet()));
```

Comment:
reducedImpacts could be initialized directly with a stream and could be a List, using Stream.distinct()

Reply:
I will keep the current code organization and keep the order of inserts in reducedImpacts intended I think